### PR TITLE
MF-792 - Users service not working in offline mode

### LINKED ISSUE #792: Users service not working in offline mode
**BUG REPORT**

1. What were you trying to achieve?
To create a user.

2. What are the expected results?
For `users` service to return `201` or `409`.

3. What are the received results?
I've received a response `400`.

4. What are the steps to reproduce the issue?
    1. Run mainflux `docker-compose -f docker/docker-compose.yml up -d`,
    2. Turn off internet connection,
    3. Try to create a user with a request like this one (it's important that domain in the email address is not `localhost` nor `example.com`):
`curl -s -S -i --cacert docker/ssl/certs/mainflux-server.crt --insecure -X POST -H "Content-Type: application/json" https://localhost/users -d '{"email":"testadmin@email.com", "password":"1234"}'`,
    4. You should receive response `400`,
    5. Turn on internet connection,
    6. Run the same request again,
    7. You should receive response `201`.

5. In what environment did you encounter the issue?
Fedora 29

6. Additional information you deem important:
I recommend that we add env var for this email validation.

### PR DESCRIPTION
**Pull request by anovakovic01**
### What does this do?
Enables `users` service to work in an offline mode.

### Which issue(s) does this PR fix/relate to?
Resolves #792.

### List any changes that modify/break current functionality
There are no such changes.

### Have you included tests for your changes?
There are already tests written for this validation.

### Did you document any new/modified functionality?
There is no new functionality.

### CODE REVIEW

**Review by nmarcetic — APPROVED** [review]

**Review by dborovcanin — APPROVED** [review]

File: users/users.go [hunk 1]
```diff
@@ -9,8 +9,14 @@ package users
 
 import (
 	"context"
+	"regexp"
+	"strings"
+)
 
-	"github.com/asaskevich/govalidator"
+var (
+	userRegexp    = regexp.MustCompile("^[a-zA-Z0-9!#$%&'*+/=?^_`{|}~.-]+$")
```
Comment (chombium) [inline users/users.go:17]: The email user (the local part) and the host (the domain part) can also have non ASCII UTF-8 characters. These  regex expressions will make them invalid.

Do we want to enable email addresses which have UTF-8 characters as well?

**Review by chombium — CHANGES_REQUESTED** [review]
We should definitely change the validation, as others are also having [problems](https://github.com/asaskevich/govalidator/issues/337). But the validation function should be adjusted.

File: users/users.go [hunk 2]
```diff
@@ -42,3 +48,27 @@ type UserRepository interface {
 	// RetrieveByID retrieves user by its unique identifier (i.e. email).
 	RetrieveByID(context.Context, string) (User, error)
 }
+
+func isEmail(email string) bool {
+	if len(email) < 6 || len(email) > 254 {
```
[the remark this answers is not in the document]
Comment (anovakovic01) [reply users/users.go:53]: @chombium these are all valid points, but I think that this should be handled in a separate PR.  I recommend that you write an issue for these validation improvements. This code was just a quick bug fix, so this code is from `govalidator` lib.

[the remark this answers is not in the document]
Comment (chombium) [reply users/users.go:53]: @anovakovic01 I've opened a new issue already #798 ;)

**Pull request merged**